theano/gof/cmodule.py

```diff
@@ -1969,6 +1974,14 @@ def compile_str(module_name, src_code, location=None,
         else:
             cmd.extend(preargs)
         cmd.extend('-I%s' % idir for idir in include_dirs)
+        if hide_symbols and sys.platform != 'win32':
+            # This has been available since gcc 4.0 so we suppose it
+            # is always available. We pass it here since it
+            # significantly reduces the size of the symbol table for
+            # the objects we want to share. This in turns leads to
+            # improved loading times on most platforms (win32 is
+            # different, as usual).
+            cmd.append('-fvisibility=hidden')
```

Comment:
Because otherwise the hide_symbols parameter would have to be on compile_args and I think it fits better with compile_str.

However I don't have that much of a strong opinion either, as long as we make sure to always pass it.